src/heretic/p_inter.c

```diff
@@ -1350,7 +1350,7 @@ void P_DamageMobj
         ang = R_PointToAngle2(inflictor->x, inflictor->y,
                               target->x, target->y);
         //thrust = damage*(FRACUNIT>>3)*100/target->info->mass;
-        thrust = damage * (FRACUNIT >> 3) * 150 / target->info->mass;
+        thrust = damage * (FRACUNIT >> 3) * 150u / target->info->mass;
```

Comment:
Well, I am sure the overflow might indeed happen in Vanilla Heretic. So, by declaring this as unsigned, you fix the bug, but you also change how Heretic works.